src/Microsoft.AspNetCore.Authentication/AuthenticationSchemeOptions.cs

```diff
@@ -36,5 +36,15 @@ public virtual void Validate()
         /// If set, will be used as the service type to get the Events instance instead of the property.
         /// </summary>
         public Type EventsType { get; set; }
+
+        /// <summary>
+        /// Used to ensure that the options are only initialized once.
+        /// </summary>
+        public bool Initialized { get; set; }
```

Comment:
Well auth is doing something a bit different than normal options usages, as handlers are doing fixup after all the Configures have been run.  If we bring back some kind of lifetime/second validation pass we could do it in options directly, but for now, we have to roll our own...